lib/internal/urlCache.js

```diff
@@ -0,0 +1,83 @@
+"use strict";
+
+
+
+function urlCache()
+{
+	var thisObj = this;
+	this.cache = {};
+	this.retrievalStore = {};
+}
+
+
+
+urlCache.prototype.clear = function()
+{
+	this.cache = {};
+	this.retrievalStore = {};
+}
+
+
+
+urlCache.prototype.store = function(url, executionFn)
+{
+	this.cache[url] = executionFn;
+
+	if (Array.isArray(this.retrievalStore[url]) === true)
+	{
+		this.retrievalStore[url].forEach( function(callback)
```

Comment:
I'd prefer for this to be a regular `for()` loop so that additional closures are not added to the garbage collector queue.